.github/workflows/test_suite_psyclone_3.1.0.yml

```diff
@@ -0,0 +1,69 @@
+name: Test suite
+
+on:
+  # Trigger the workflow on certain events
+  pull_request:
+    branches:
+      - "psyclone-3.1.0"
+  push:
+    branches:
+      - "psyclone-3.1.0"
```

Comment:
Please could you open a separate test PR that would merge into the `psyclone-3.1.0` branch so that we can test these get triggered, as expected?